### ISSUE TITLE
New-NetNat support check not working as intended

### ISSUE THREAD

**Issue by Dola-Shuvi** [micahmo/WireGuardServerForWindows#62]
First off let me thank you for making this application. I only found it yesterday but managed to get a WireGuard server running in under an hour.

Now onto the issue at hand.

In commit eb22d730ab70ae5638a010128a70841d5e4464fe a support check for New-NetNat was introduced. This support check fails on my machine even though New-NetNat is supported. Running the support check in powershell yields the following output:
```
PS C:\WINDOWS\system32> Get-Help New-NetNat -Parameter InternalIPInterfaceAddressPrefix
Get-Help : Kein Parameter entspricht dem Kriterium "InternalIPInterfaceAddressPrefix".
In Zeile:1 Zeichen:1
+ Get-Help New-NetNat -Parameter InternalIPInterfaceAddressPrefix
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (System.Manageme...CommandHelpInfo:ProviderCommandHelpInfo) [Get-Help],
    PSArgumentException
    + FullyQualifiedErrorId : NoParmsFound,Microsoft.PowerShell.Commands.GetHelpCommand
 ```
Get-Help fails to find information about the InternalIPInterfaceAddressPrefix parameter and gives an exit code other than 0. Running the commands to manually set up NAT routing works fine however and no errors occur. After rewriting the support check and letting it do the setup everything worked fine aswell. I have pushed my changes to https://github.com/Dola-Shuvi/WgServerforWindows/commit/41413db1a86a4dbde74a3cba317c761da64b4f8e for you to take a look at.

My Windows version is 22H2 (Build 19045.2130) and I installed everything except the Hyper-V-Hypervisor and the GUI manager in the add/remove Windows features menu.

**Comment by micahmo** [comment]
Hey @Dola-Shuvi! Thanks so much for taking the time to report this issue and write a potential fix.

I am a bit hesitant to make changes with side-effects just to perform a test. What if the remove fails? Now the user has `testnat` on their system forever. It may also clobber the `172.22.0.0/24` subnet, if that's already configured on the user's system.

I think the more interesting question is why `Get-Help` fails on your system, despite your version of `New-NetNat` clearly supporting the `InternalIPInterfaceAddressPrefix` parameter.

I did some testing and found that, somehow, downloading the help files makes things worse! Check this out.

``` powershell
PS C:\Windows\system32> Get-Help New-NetNat -Parameter InternalIPInterfaceAddressPrefix
-InternalIPInterfaceAddressPrefix <string>

    Required?                    false
    Position?                    Named
    Accept pipeline input?       false
    Parameter set name           cim:CreateInstance0
    Aliases                      None
    Dynamic?                     false




PS C:\Windows\system32> update-help netnat
PS C:\Windows\system32> Get-Help New-NetNat -Parameter InternalIPInterfaceAddressPrefix
Get-Help : No parameter matches criteria InternalIPInterfaceAddressPrefix.
At line:1 char:1
+ Get-Help New-NetNat -Parameter InternalIPInterfaceAddressPrefix
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (System.Manageme...CommandHelpInfo:ProviderCommandHelpInfo) [Get-Help],
    PSArgumentException
    + FullyQualifiedErrorId : NoParmsFound,Microsoft.PowerShell.Commands.GetHelpCommand
 
```

It may also be related to [this issue with German systems](https://devblogs.microsoft.com/powershell/get-help-online-fails-in-german/).

Would you be willing to try a few more things?
* I wonder if `Get-Help New-NetNat -Parameter *` would work. If so, we could run that and simply check the output for `InternalIPInterfaceAddressPrefix`.
* Maybe just run `New-NetNat -InternalIPInterfaceAddressPrefix` and ensure the output has `Missing an argument for parameter 'InternalIPInterfaceAddressPrefix'`. I assume if the parameter doesn't exist on the version of Windows, the output would be different (we'd need to verify).
* Maybe there some other version (cmdlet, Windows) that would tell us with certainty whether the parameter exists. To be honest, I don't remember when it doesn't, but I definitely ran a test where the command existed and the parameter didn't, which is why I added that extra check.

Let me know what you think.

**Comment by micahmo** [comment]
> I found this small query `(Get-Command New-NetNat).Parameters['InternalIPInterfaceAddressPrefix']` over on [StackOverflow ](https://stackoverflow.com/a/71856199)

Ooh this is great, nice find! Relying on `Get-Help` was probably always a bad idea. 😄

Unfortunately, that command returns `0` whether or not the parameter exists, but this seems to do the trick.

``` powershell
if ((Get-Command New-NetNat).Parameters.ContainsKey('InternalIPInterfaceAddressPrefix')) { exit 0 } else { exit 1 }
```

Would you be willing to test it for me before I make a new release? I've pushed the change to [fix/62](https://github.com/micahmo/WgServerforWindows/compare/main...fix/62). I can also generate a build for you, but it sounds like you know how to do that. 😊

**Comment by micahmo** [comment]
> Ideally we should also keep the old support check as a fallback

Good point! Would be willing to take a peek at my PR #63?

**Comment by micahmo** [comment]
Fixed in [v2.0.3](https://github.com/micahmo/WgServerforWindows/releases/tag/v2.0.3).
